src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorRuntimeNodeWriter.cs

```diff
@@ -33,14 +33,14 @@ internal class BlazorRuntimeNodeWriter : BlazorNodeWriter
 
         private readonly ScopeStack _scopeStack = new ScopeStack();
         private string _unconsumedHtml;
-        private IList<object> _currentAttributeValues;
+        private List<IntermediateToken> _currentAttributeValues;
         private IDictionary<string, PendingAttribute> _currentElementAttributes = new Dictionary<string, PendingAttribute>();
         private IList<PendingAttributeToken> _currentElementAttributeTokens = new List<PendingAttributeToken>();
         private int _sourceSequence = 0;
 
         private struct PendingAttribute
         {
-            public object AttributeValue;
+            public List<IntermediateToken> Values { get; set; }
```

Comment:
OK so what I did here was just treat everything as tokens since we already have that information. This makes the tracking of what's what very simple because it's all tokens.

Reply:
Sounds good